docs/docs/reference/gen_notebooks/log_autogen_group_chat.md

```diff
@@ -0,0 +1,212 @@
+
+:::
+
+
+<!--- @wandbcode{log-autogen-group-chat} -->
+
+# Log AutoGen Group Chat
+
+[AutoGen](https://microsoft.github.io/autogen/0.2/) is an innovative open-source framework designed to streamline the development of AI agents capable of cooperative task-solving. Key features include support for large language models (LLMs), agent conversation capabilities, and both autonomous and human-assisted workflows. AutoGen is essential for building next-generation LLM applications, simplifying the orchestration and optimization of complex multi-agent conversations and boosting the performance of LLMs.
+
+The Group Chat feature of AutoGen allows developers to customize agent interactions by defining conversation patterns and workflows. This customization is crucial as it enables precise control over the dialogue dynamics among agents, making development less cumbersome and more efficient. Tracking these interactions is vital to refining the agent's performance and ensuring the successful deployment of sophisticated multi-agent systems.
+
+This tutorial demonstrates how we can build multi-agent system with AutoGen Group Chat and trace them with [Weave](../../introduction.md).
+
+:::tip[NOTE]
+This example refers to an example in [AutoGen documentation](https://microsoft.github.io/autogen/0.2/docs/notebooks/agentchat_groupchat_research/)
+:::
+
+## Installing the Dependencies
+
+We need the following libraries for this tutorial:
+- [AutoGen](https://microsoft.github.io/autogen/0.2/) to create multi-agent system.
+- [Weave](../../introduction.md) to track our LLM workflow and evaluate our prompting strategies.
+
+```python
+!pip install -qU autogen-agentchat~=0.2 weave wandb
+```
+
+We set `WANDB_API_KEY` in our environment variable so that we may easily login with `wandb.login()` (this should be given to the colab as a secret).
+We set the project in W&B we want to log this into in `name_of_wandb_project`.
+
+**NOTE**: `name_of_wandb_project` may also be in the format of `{team_name}/{project_name}` to specify a team to log the traces into.
+We then fetch a weave client by calling weave.init()
+
+Since we'll be using the [OpenAI API](https://openai.com/index/openai-api/), we will also need an OpenAI API key. You can [sign up](https://platform.openai.com/signup) on the OpenAI platform to get your own API key. (this should be given to the colab as a secret too)
+
+```python
+import os
+import autogen
+
+from google.colab import userdata
+
+import weave
+from weave import Model
+import wandb
+
+os.environ["WANDB_API_KEY"] = userdata.get("WANDB_API_KEY")
+os.environ["WANDB_API_KEY"] = userdata.get("OPENAI_API_KEY")
```

Comment:
Typo, should be OPENAI_API_KEY

Reply:
Sorry, this is my mistake.  Will update after we develop an integration with Autogen.